packages/tupaia-web/src/features/Visuals/Matrix/parseData.tsx

```diff
@@ -121,27 +133,51 @@ export const parseRows = (
 
       if (!matchesSearchFilter) return result;
     }
-    // otherwise, handle as a regular row
-    result.push({
-      title: dataElement,
+
+    const newResult = {
```

Comment:
```suggestion
    const newResult: MatrixRowType = {
```
I think this is technically safer because it doesn't cast the type, it tells TS that this object is this type and will tell us if anything doesn't match the type